samples/cellular/modem_shell/src/fota/fota.c

```diff
@@ -16,58 +16,54 @@
 #include "link.h"
 #include "mosh_print.h"
 
-static void modem_update_apply(void)
+static void on_modem_lib_dfu(int dfu_res, void *ctx)
 {
-	int err;
-
-	link_func_mode_set(LTE_LC_FUNC_MODE_OFFLINE, true);
-
-	mosh_print("FOTA: Shutting down modem to trigger DFU update...");
-
-	err = nrf_modem_lib_shutdown();
-	if (err) {
-		mosh_warn("FOTA: Failed to shut down modem, err: %d", err);
-	}
-
-	err = nrf_modem_lib_init();
-	switch (err) {
-	case 0:
-		mosh_error("FOTA: Expected DFU result from modem library, "
-			   "but no DFU was triggered");
-		goto exit;
+	switch (dfu_res) {
 	case NRF_MODEM_DFU_RESULT_OK:
-		mosh_print("FOTA: Modem firmware update successful!");
-		goto restart_modem;
+		mosh_error("Modem firmware updated");
+		mosh_error("FOTA: Modem firmware update successful!");
```

Comment:
Why two similar prints?
@tokangas

Reply:
I'll remove the first one.

Reply:
Also, the function has changed from `mosh_print()` to `mosh_error()`. It should be `mosh_print()`, because this is not an error.